diff --git a/node_modules/@discordjs/voice/dist/index.js b/node_modules/@discordjs/voice/dist/index.js
index 2f2e420..6f9eb08 100644
--- a/node_modules/@discordjs/voice/dist/index.js
+++ b/node_modules/@discordjs/voice/dist/index.js
@@ -310,8 +310,9 @@ var VoiceUDPSocket = class extends import_node_events.EventEmitter {
   }
   keepAlive() {
     if (this.keepAlives.length >= KEEP_ALIVE_LIMIT) {
+      this.keepAlives = []
       this.debug?.("UDP socket has not received enough responses from Discord - closing socket");
-      this.destroy();
+      //this.destroy();
       return;
     }
     this.keepAliveBuffer.writeUInt32LE(this.keepAliveCounter, 0);
