diff --git a/node_modules/ytdl-core/lib/utils.js b/node_modules/ytdl-core/lib/utils.js
index 143ef1f..cbe1d52 100644
--- a/node_modules/ytdl-core/lib/utils.js
+++ b/node_modules/ytdl-core/lib/utils.js
@@ -98,6 +98,12 @@ exports.cutAfterJS = mixedJson => {
   for (i = 0; i < mixedJson.length; i++) {
     // End of current escaped object
     if (!isEscaped && isEscapedObject !== null && mixedJson[i] === isEscapedObject.end) {
+      // Ignore character inside RegEx character set with a range of 30 characters
+      let charsetStart = mixedJson.substring(i - 15, i).search(/\[[^\]]*/)
+      let charsetEnd = mixedJson.substring(i, i + 15).search(/[^\[]*\]/)
+      if (mixedJson[i] == '/' && charsetStart !== -1 && charsetEnd !== -1)
+        continue;
+
       isEscapedObject = null;
       continue;
     // Might be the start of a new escaped object
